query-engine/connectors/mongodb-query-connector/src/root_queries/write.rs

```diff
@@ -53,6 +54,8 @@ pub async fn create_record<'conn>(
         doc.insert(field.db_name().to_owned(), bson);
     }
 
+    logger::log_insert_one(coll.name(), &doc);
```

Comment:
I don't really like this solution as opposed to the read builder that automatically logs the query. It's super error prone to forget logging. For this initial PR it's okay, but in the future I'd rather wrap all write operations like the read operations.

Reply:
yes, but it is not possible with the way _the code is done_. The way you did the read operations was to abstract them in a single Query handler operation (difference between `FindOne` and `FindMany`, in the insert operations you have a plethora of small functions doing a single thing and the only place where you will know such thing is in the function itself. That proposed change will require _recoding the whole insert/update/delete operation to bring consistency_ and that was beyond this logging stuff...

Reply:
You'd have to elaborate on "the way the code is done" more. Not right now, but when we decide to change it.